src/model_repository_manager.cc

```diff
@@ -2162,6 +2220,67 @@ ModelRepositoryManager::UpdateDependencyGraph(
   return Status::Success;
 }
 
+Status
+ModelRepositoryManager::AddModelRepositoryPath(const std::string& path)
+{
+  if (!repository_paths_.insert(path).second) {
```

Comment:
Need mutex to protect concurrent calls to the function here and below`std::lock_guard<std::mutex> lock(poll_mu_);`

Reply:
Good catch, updated.